src/assets/src/hooks/useWebSocket.ts

```diff
@@ -1,19 +1,21 @@
 import { useState, useEffect } from "react";
+import ReconnectingWebSocket from 'reconnecting-websocket';
 
 interface OfficeHoursMessage<T> {
     type: "init"|"update"|"deleted";
     content: T;
 }
 
-const closeCodes = {
-    1006: "An unexpected error occurred. Please refresh the page.",
-    4404: "The resource you're looking for could not be found. Maybe it was deleted?",
-} as {[closeCode: number]: string}
-
 export const useWebSocket = <T>(url: string, onUpdate: (content: T) => void, onDelete?: (setError: (React.Dispatch<React.SetStateAction<Error | undefined>>)) => void) => {
     const [error, setError] = useState(undefined as Error | undefined);
     useEffect(() => {
-        const ws = new WebSocket(url);
+        const ws = new ReconnectingWebSocket(
+            url,
+            undefined,
+            {
+                maxRetries: 3,
+            },
```

Comment:
It's also tested within the context of the library itself, but within the context of this app, it will retry three times and then stop doing work. The error message doesn't change. It could be changed after x retries to something like "Retries failed, please refresh or contact the ITS Service Center." But I think this treads into the realm of very unlikely edge-cases.

Reply:
I would agree, I was just wondering if you tried to simulate such a failure to see how the library worked.

Reply:
I didn't simulate it per se, I just let it time out naturally by disconnecting my internet.

Reply:
Cool, yeah, I just did something similar by stopping the `web` container. Seems to have stopped after three attempts. I'll mark this as resolved.